content/docs/command-reference/exp/run.md

```diff
@@ -69,27 +69,27 @@ checkpoints with DVC during your code or script runtime (similar to a logger).
 
 To do so, first mark stage `outs` with `checkpoint: true` in `dvc.yaml`. Having
 at least a checkpoint <abbr>output</abbr> is needed so that the experiment can
-later restart based on that output's last <abbr>cached</abbr> state.
+later continue from that output's last <abbr>cached</abbr> state.
 
 ⚠️ Using the `checkpoint` field in `dvc.yaml` is only compatibly with
 `dvc exp run`, `dvc repro` will abort if any stage contains it.
 
-Then, in the corresponding code, either call the `dvc.api.make_checkpoint()`
-function (Python), or write a signal file (any programming language) following
-the same steps as `make_checkpoint()` — please refer to that reference for
-details.
+Then, in your code either call the `dvc.api.make_checkpoint()` function
+(Python), or write a signal file (any programming language) following the same
+steps as `make_checkpoint()` — please refer to its reference for details.
 
-Once this is setup, you can use `dvc exp run` to begin the experiment. When the
-process finishes or gets interrupted (e.g. with Ctrl + `C`), DVC will
-[apply](/doc/command-reference/exp/apply) the last checkpoint to the
-<abbr>workspace</abbr> (overwriting any further changes done by the stage).
+You can now use `dvc exp run` to begin the experiment. This removes any
+`checkpoint` outputs before running the experiment (regardless of whether they
+have cached versions). When the process finishes or gets interrupted (e.g. with
+Ctrl + `C`), DVC will [apply](/doc/command-reference/exp/apply) the last
+checkpoint to the <abbr>workspace</abbr> (overwriting any further changes done
+by the stage).
```

Comment:
```suggestion
checkpoint to the <abbr>workspace</abbr> (overwriting any further changes done
by the stage if the process was interrupted).
```
Alternatively:
```
When the process finishes, DVC will [apply](/doc/command-reference/exp/apply) the last
checkpoint to the <abbr>workspace</abbr>. If the process gets interrupted  (e.g. with
Ctrl + `C`), DVC will overwrite any changes done by the stage after the last checkpoint.
```
I find "overwriting any further changes" confusing unless it's specifically tied to the process being interrupted. If the process isn't interrupted, I think a final checkpoint is automatically generated to save the final outcome, so it's not overwriting further changes (@pmrowla can you confirm if that's correct?).

Reply:
Yeah, there's an additional commit generated at the end of the stage (assuming anything changed since the last time the user called `make_checkpoint`), and that is what gets applied. Even if the process is interrupted, we don't overwrite anything in the user's workspace.

Reply:
So should `(overwriting any further changes done by the stage)` get dropped?

Reply:
> there's an additional commit generated at the end of the stage

Ah this I didn't realize. OK I rewrote this based on that and Dave's suggestions (thanks) for now.

> should (overwriting any further changes done by the stage) get dropped?

Woah wo why did I think that? A bit confused on this, so a final checkpoint that reflects the resulting workspace "is applied" (what is there to apply?) but the last checkpoint for interrupted runs is NOT applied?

Reply:
Here's the latest language for reference:

```
If the process gets
interrupted (e.g. with Ctrl + `C`), DVC will
[apply](/doc/command-reference/exp/apply) the last checkpoint to the
<abbr>workspace</abbr> (overwriting any further changes).
```

Pending confirmation from @pmrowla, maybe we should remove any reference to `apply` or the workspace if nothing gets overwritten? Maybe this sentence isn't needed at all? Do we feel it necessary to call out process interruption to make explicit that this is an expected workflow? If so, maybe something like:

```
If the process gets
interrupted (e.g. with Ctrl + `C`), DVC will
keep all checkpoints recorded before interruption.
```

Reply:
OK. Removed `apply` and workspace references for now per Peter's comment above but it would def. be great to confirm!